### DISCUSSION TITLE
Static word embeddings search by word

### DISCUSSION THREAD

**Discussion by djstrong** [milvus-io/milvus#27712] [Q&A and General discussion] [answered]
Could you tell me what is the best index for static word embeddings where search argument is not a vector but one of the words? For example in HNSW (graph based index) there is no need for search the closest embedding because we know the node.

**Comment by yhmo** [comment]
By Milvus you can:
- store embeddings
- search embedding by embedding

Milvus search is approximate/similar. Elastic is the right choice if you want to do a keyword search.

**Reply by djstrong** [reply]
I want to search similar embeddings by embedding present in the index (knowing exact node). If you know how HNSW works then it might be not optimal.

**Reply by codingjaguar** [reply, marked as answer]
There is no support for "fixed" embedding in Milvus. Even word embedding has a finite list of vectors (aka the number of tokens in the vocabulary), there is no such "more optimal way" than searching with a random vector.

Unlike reverted index in full text search, vector database doesn't build reverse mapping from an exact vector.

**Reply by djstrong** [reply]
Ok, so Milvus does not support this case in special, optimal way. Thank you.
